src/tlsserial/happy_certificate.py

```diff
@@ -0,0 +1,209 @@
+
+NAME_ATTRIBS = [
+    ("CN", NameOID.COMMON_NAME),
+    ("O", NameOID.ORGANIZATION_NAME),
+    ("OU", NameOID.ORGANIZATIONAL_UNIT_NAME),
+    ("L", NameOID.LOCALITY_NAME),
+    ("ST", NameOID.STATE_OR_PROVINCE_NAME),
+    ("C", NameOID.COUNTRY_NAME),
+    ("DC", NameOID.DOMAIN_COMPONENT),
+    ("E", NameOID.EMAIL_ADDRESS),
+]
+
+@dataclass
+class HappyCertificate:
+    """HappyCertificate is a happy certificate."""
+
+    cert: x509.Certificate
+
+
+    @property
+    def version(self) -> int:
+        """Return the x509 version."""
+        return self.cert.version.value + 1
+
+    @property
+    def issuer(self) -> list[str]:
+        """Issuer."""
+        return [
+            f"[{a}] {n.value}"
+            for a, b in NAME_ATTRIBS
+            for n in self.cert.issuer.get_attributes_for_oid(b)
+        ]
+
+    @property
+    def subject(self) -> List[str]:
+        """Subject."""
+        return [
+            f"[{a}] {n.value}"
+            for a, b in NAME_ATTRIBS
+            for n in self.cert.subject.get_attributes_for_oid(b)
+        ]
+
+    @property
+    def sans(self) -> list[Any]:
+        """The Subject Alternative Names."""
+        try:
+            return self.cert.extensions.get_extension_for_oid(
+                ExtensionOID.SUBJECT_ALTERNATIVE_NAME
+            ).value.get_values_for_type(x509.DNSName)
+        except ExtensionNotFound:
+            return []
+
+    @property
+    def basic_constraints(self) -> Dict[str, str]:
+        """Return the CA BasicConstraint properties."""
+        basic_constraints: Dict[str, str] = {}
+        try:
+            basic_constraints_object = self.cert.extensions.get_extension_for_oid(
+                ExtensionOID.BASIC_CONSTRAINTS
+            ).value
+            basic_constraints["ca"] = str(basic_constraints_object.ca)
+            basic_constraints["path_length"] = str(
+                basic_constraints_object.path_length
+            )
+        except (ValueError, ExtensionNotFound):
+            pass
+        return basic_constraints
+
+    @property
+    def key_usage(self) -> list[str]:
+        """Key usage."""
+        key_usage = []
+        try:
+            key_usage_object = self.cert.extensions.get_extension_for_oid(
+                ExtensionOID.KEY_USAGE
+            ).value
+            for attr, value in key_usage_object.__dict__.items():
+                if value is True:
+                    key_usage.append(attr.lstrip("_"))
+        except (UnboundLocalError, ValueError, ExtensionNotFound):
+            pass
+        return key_usage
+
+    @property
+    def ext_key_usage(self) -> list[str]:
+        """Returns list of Extended key usages."""
+        ext_key_usage = []
+        try:
+            ext_key_usage_object = self.cert.extensions.get_extension_for_oid(
+                ExtensionOID.EXTENDED_KEY_USAGE
+            ).value
+            for usage in ext_key_usage_object.__dict__["_usages"]:
```

Comment:
Is there a way I can avoid this calling of dunder methods with private sounding key names?